namespaces/openstack.yml

```diff
@@ -0,0 +1,120 @@
+version: 2016.03.10.0
+
+defaults:
+  type: string
+  doc_values: true
+  index: not_analyzed
+
+namespace:
+  name: openstack
+  type: group
+  description: >
+    Mapping specifically for openstack. Inherits all the fields from
+    __default__.
+  fields:
+
+    - name: component_name
+      type: string
+      description: >
+        The name of the component that produced this log message.  For
+        example, "neutron.agent.linux.utils" or "nova.compute.api" or
+        "nova.policy".
+      norms:
+        enabled: True
+
+    # BEGIN oslo context
+    #
+    # Most openstack services log a standard context string that
+    # includes the follow fields.  See the defaults for
+    # logging_context_format_string and logging_user_identity_format
+    # in http://docs.openstack.org/developer/oslo.log/opts.html
+
+    - name: request_id
+      type: string
+      description: >
+        TODO.
+      norms:
+        enabled: True
+
+    - name: user_uuid
+      type: string
+      description: >
+        TODO.
+      norms:
+        enabled: True
+
+    - name: project_uuid
+      type: string
+      description: >
+        TODO.
+      norms:
+        enabled: True
+
+    - name: domain
+      type: string
+      description: >
+        TODO.
+      norms:
+        enabled: True
+
+    - name: user_domain
+      type: string
+      description: >
+        TODO.
+      norms:
+        enabled: True
+
+    - name: project_domain
+      type: string
+      description: >
+        TODO.
+      norms:
+        enabled: True
+
+    # END oslo context
+
+    - name: nova_server_uuid
```

Comment:
A question I have for folks reviewing this: there are ultimately a *lot* of resource uuids we can in theory extract from openstack (e.g., heat stacks, heat resources, keystone users/projects/domains/groups, mistral workflows/actions/environments/tasks/workbooks/etc, and so forth).  Should we?  Or should we just stick with the common metadata for now (which would mostly be the stuff above the `END oslo context` line above)?